numpy/core/multiarray.pyi

```diff
@@ -91,7 +91,7 @@ _DTypeLike = Union[
     Type[_SCT],
     _SupportsDType[dtype[_SCT]],
 ]
-_ArrayLike = _NestedSequence[_SupportsArray[dtype[_SCT]]]
+_ArrayLike = _FiniteNestedSequence[_SupportsArray[dtype[_SCT]]]
```

Comment:
This is used with typevars in `def func(a: _ArrayLike[_T]) -> NDArray[_T]: ...`-esque expressions, so unfortunately we can't use the new fancy protocol here.